scan-command/src/test/java/io/ballerina/scan/internal/StaticCodeAnalyzerTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.scan.internal;
+
+import io.ballerina.projects.Document;
+import io.ballerina.projects.Module;
+import io.ballerina.projects.Project;
+import io.ballerina.projects.directory.SingleFileProject;
+import io.ballerina.scan.BaseTest;
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.Rule;
+import io.ballerina.scan.RuleKind;
+import io.ballerina.scan.Source;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * Core analyzer tests.
+ *
+ * @since 0.1.0
+ */
+public class StaticCodeAnalyzerTest extends BaseTest {
+    private final Path coreRuleBalFiles = testResources.resolve("test-resources").resolve("core-rules");
+
+    @Test(description = "test checkpanic analyzer")
+    void testCheckpanicAnalyzer() {
+        String ballerinaFile = "rule_checkpanic.bal";
+        Project project = SingleFileProject.load(coreRuleBalFiles.resolve(ballerinaFile));
+        Module defaultModule = project.currentPackage().getDefaultModule();
+        Document document = defaultModule.document(defaultModule.documentIds().iterator().next());
```

Comment:
I think this code snippet will be reused when we add more tests. Shall move it to a another method?
```suggestion
        Project project = SingleFileProject.load(coreRuleBalFiles.resolve(ballerinaFile));
        Module defaultModule = project.currentPackage().getDefaultModule();
        Document document = defaultModule.document(defaultModule.documentIds().iterator().next());
```